hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/spec/RestSpecTest.java

```diff
@@ -0,0 +1,131 @@
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.json.JSONException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestClient;
+import org.testcontainers.containers.GenericContainer;
+
+@RequiredArgsConstructor
+@RunWith(SpringRunner.class)
+@EnableConfigurationProperties(value = RestJavaProperties.class)
+@SpringBootTest(
+        classes = SpecTestConfig.class,
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+        properties = {"spring.main.allow-bean-definition-overriding=true"})
+public class RestSpecTest extends RestJavaIntegrationTest {
+    private static final int JS_REST_API_CONTAINER_PORT = 5551;
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private static final List<File> SELECTED_SPECS = List.of(
+            new File("../hedera-mirror-rest/__tests__/specs/accounts/alias-into-evm-address.json"),
+            new File("../hedera-mirror-rest/__tests__/specs/blocks/no-records.json"),
+            new File("../hedera-mirror-rest/__tests__/specs/accounts/specific-id.json")
+    );
+
+    @Resource
+    private SpecDomainBuilder specDomainBuilder;
+
+    @Autowired
+    GenericContainer<?> jsRestApi;
+
+    @Autowired
+    private RestJavaProperties properties;
+
+    @LocalServerPort
+    private int port;
+
+    private RestClient.Builder restClientBuilder;
+
+    private String baseJsRestApiUrl;
+
+    @BeforeEach
+    final void setup() {
+        baseJsRestApiUrl = "http://%s:%d".formatted(jsRestApi.getHost(), jsRestApi.getMappedPort(JS_REST_API_CONTAINER_PORT));
+        log.info("setup - baseJsRestApiUrl: {}", baseJsRestApiUrl);
+        restClientBuilder = RestClient.builder()
+                .defaultHeader("Accept", "application/json")
+                .defaultHeader("Access-Control-Request-Method", "GET")
+                .defaultHeader("Origin", "http://example.com");
+    }
+
+    @Test
+    void runSelectedSpecs() {
+        SELECTED_SPECS.forEach(this::runSpecTests);
+    }
+
+    private void runSpecTests(File specFile) {
+        RestSpec restSpec;
+        try {
+            restSpec = OBJECT_MAPPER.readValue(specFile, RestSpec.class);
+        } catch (IOException e) {
+            log.warn("Failed to parse spec file: {}", specFile, e);
+            return;
+        }
+
+        var normalizedRestSpec = RestSpecNormalized.from(restSpec);
```

Comment:
Consolidate logic related to loading specs to a single method.

```suggestion
        var restSpec = loadSpecs();
```

Reply:
Using `@ParameterizedTest`, I have redone this logic.